app/models/survey/poll.rb

```diff
@@ -15,9 +15,19 @@ class Survey::Poll < ApplicationRecord
 
   before_save :set_visibility_by_role
 
-  # TODO: real scopes based on `date` instead of `visible`
-  scope :ongoing, -> { where(visible: true) }
-  scope :archived, -> { where(visible: false) }
+  # Startdatum liegt in der Vergangenheit oder Heute und Enddatum liegt in der Zukunft oder Heute:
+  scope :ongoing, lambda {
+    where.not(fixed_dates: { date_start: [nil, ""] })
+      .where.not(fixed_dates: { date_end: [nil, ""] })
+      .where("fixed_dates.date_start <= ? AND fixed_dates.date_end >= ?", Date.today, Date.today)
+      .joins(:date)
+  }
+
+  # Enddatum liegt in der Vergangenheit:
+  scope :archived, lambda {
+    where.not(fixed_dates: { date_end: [nil, ""] })
+      .where("fixed_dates.date_end < ?", Date.today).joins(:date)
+  }
```

Comment:
die scopes funktionieren gut in meinen tests. yea!

aber hab festgestellt, dass der id filter für `all` noch nicht rund ist. wenn keine id mitgegeben wir, wie ich es im insomnia habe, dann krieg ich n leeres array.

```
query Surveys($id: ID) {
  all: surveys(ids: [$id]) {
    ...surveyFields
  }
  ...
```

das liegt daran, dass hier in der query id ein leeres array ist.
wir brauchen also noch folgenden check, würd ich sagen:

```
def apply_ids(scope, value)
  scope.where(id: value) if value.present? && value.any?
end
```

wir können ja nicht ausschließen, dass jemand sonst per leerem array anfragt, wobei die erwartung denk ich schon ist, dass dann alle returned werden?!

Reply:
wenn wir die möglichkeit haben da auch einfach gar kein array reinzugeben und er filtert dann nicht, sondern gibt alle, dann reicht das.
wenn jemand auf ein leeres array filtert und nichts zurückbekommt, ist ja sogesehen nichts kaputt. "zeige mir alle, einträge die eine id aus folgendem array haben" wird ja erledigt

Reply:
```
query Surveys {
  all: surveys {
    ...surveyFields
  }
}
```

geht natürlich ja. dann braucht es dafür aber eine extra query ohne möglichen parameter. es kann dann nicht mehr alles über dieselbe query abgehandelt werden. können wir natürlich wissen und überall so machen. dann braucht es keine zusätzliche prüfung auf vorhandensein von ids.

Reply:
hmmm... bei den anderen Revolvern haben wir auch keine Prüfung auf das Vorhandensein der ID. Und ich gebe @digorath Recht dass es komisch wirkt,  wenn ich ein leeres Array übergebe und alle Einträge bekomme.

Reply:
Alternativ wäre, wir bauen noch eine 2 Query wie bei den anderen speziell um einen einzelnen Datensatz abzurufen:

```
field :news_items, function: Resolvers::NewsItemsSearch
field :news_item, QueryTypes::NewsItemType, null: false do
      argument :id, ID, required: true
end
```

dann könnte man auch den ID Filter komplett entfernen

Reply:
ist aber eigentlich überflüssig, glaube ich

Reply:
jo. bin überzeugt, dass man mit der richtigen anfrage umgeht und es die änderung nicht braucht.